src/lexer/disambiguateTokens.ts

```diff
@@ -53,14 +53,32 @@ const identToArrayIdent = (token: Token, i: number, tokens: Token[]): Token => {
 
 const keywordToArrayKeyword = (token: Token, i: number, tokens: Token[]): Token => {
   if (token.type === TokenType.RESERVED_KEYWORD) {
-    const nextToken = tokens[i + 1];
+    const nextToken = nextNonCommentToken(tokens, i);
     if (nextToken && isOpenBracket(nextToken)) {
       return { ...token, type: TokenType.ARRAY_KEYWORD };
     }
   }
   return token;
 };
 
+const prevNonCommentToken = (tokens: Token[], index: number): Token | undefined =>
+  nextNonCommentToken(tokens, index, -1);
+
+const nextNonCommentToken = (
+  tokens: Token[],
+  index: number,
+  dir: -1 | 1 = 1
+): Token | undefined => {
+  let i = 1;
```

Comment:
can be `tokens.slice(index).find(token => !isComment(token))`

Reply:
`slice()` has linear complexity. The function itself is called in a loop over all tokes, which would lead to square complexity in the worst case. I'd rather avoid that.

Though it might be that JS engine optimizes away the overhead of creating a slice. Would need to be measured to be sure.

Reply:
Ran the performance comparison which showed there being no real performance difference.

However I'll stick to the original implementation as I need the function to work for searching both forward and backward through token list and the proposed simplification only works for forward-search.